src/Engine/Lod/Lod.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "Version.h"
+
+
+namespace Lod {
+enum class OpenMode {
+    Read,
+    ReadWrite,
+};
+
+
+class Lod {
+public:
+    /**
+     * Open an existing LOD file.
+     *
+     * This function loads an existing LOD file for either reading or reading+writing.
+     *
+     * @param filename          LOD filename to open.
+     * @param mode              Opening mode.
+     */
+    static std::shared_ptr<Lod> open(const std::string &filename, OpenMode mode);
```

Comment:
Why do we have factory methods? Is this class supposed to be a virtual base for several different implementations?

Also I suggest we don't use `std::shared_ptr` in factory methods because it's radioactive mud. Once you have a `shared_ptr`, you're stuck with it. Factory methods that allocate should just return `std::unique_ptr`, thus conveying the intent ("it's up to the user to deallocate this memory with operator delete"), and not mixing it up with ownership. If the user wants, he can always create a `shared_ptr` out of it.

I'd rant further and say that `std::shared_ptr` is almost always a bad thing and what you usually need is `std::unique_ptr` and organizing your code in a way that you know who owns what, but this is a long talk that touches on a lot of topics.